src/Symfony/Component/Cache/Adapter/Client/MemcachedClient.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter\Client;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedClient
```

Comment:
Marked `@internal` for now. I **really, really, really** want to avoid shoving all this logic into a single `createConnection` method within the Adapter itself...